cmd/must-gather-clean/root.go

```diff
@@ -29,31 +31,48 @@ var rootCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		defer klog.Flush()
 
-		err := cli.Run(ConfigFile, InputFolder, OutputFolder, DeleteOutputFolder, ReportingFolder, WorkerCount)
-		if err != nil {
-			klog.Exitf("%v\n", err)
+		if PipeModeEnabled {
+			err := cli.RunPipe(ConfigFile, os.Stdin, os.Stdout)
+			if err != nil {
+				klog.Exitf("%v\n", err)
+			}
+		} else {
+			err := cli.Run(ConfigFile, InputFolder, OutputFolder, DeleteOutputFolder, ReportingFolder, WorkerCount)
+			if err != nil {
+				klog.Exitf("%v\n", err)
+			}
 		}
 	},
 }
 
-func init() {
+func initFlags() {
 	flags := rootCmd.Flags()
 	flags.StringVarP(&ConfigFile, "config", "c", "", "The path to the obfuscation configuration")
-	_ = rootCmd.MarkFlagRequired("config")
 	flags.StringVarP(&InputFolder, "input", "i", "", "The directory of the must-gather dump")
-	_ = rootCmd.MarkFlagRequired("input")
 	flags.StringVarP(&OutputFolder, "output", "o", "", "The directory of the obfuscated output")
-	_ = rootCmd.MarkFlagRequired("output")
 	flags.BoolVarP(&DeleteOutputFolder, "overwrite", "d", false, "If the output directory exists, setting this flag will delete the folder and all its contents before cleaning.")
 	flags.IntVarP(&WorkerCount, "worker-count", "w", runtime.NumCPU(), "The number of workers for processing")
 	flags.StringVarP(&ReportingFolder, "report", "r", ".", "The directory of the reporting output folder, default is the current working directory")
 
+	if !PipeModeEnabled {
+		_ = rootCmd.MarkFlagRequired("config")
+		_ = rootCmd.MarkFlagRequired("input")
+		_ = rootCmd.MarkFlagRequired("output")
+	}
+
 	fs := goflag.NewFlagSet("", goflag.ExitOnError)
 	klog.InitFlags(fs)
 	rootCmd.Flags().AddGoFlagSet(fs)
 }
 
 func main() {
+	stat, _ := os.Stdin.Stat()
+	if (stat.Mode() & os.ModeCharDevice) == 0 {
```

Comment:
I don't know the details, just want to make sure bitwise AND is on purpose here?

Reply:
yep: https://stackoverflow.com/questions/43947363/detect-if-a-command-is-piped-or-not